### ISSUE TITLE
[Repo]: Test apply and then rerun apply to ensure there isn't permadrift across resources

### ISSUE THREAD

**Issue by nitrocode** [hashicorp/terraform-provider-aws#28856]
### Description

We should ensure that resources, newly created and current ones, do not have permadrift after a successful apply.

### References

- https://github.com/hashicorp/terraform-provider-aws/issues/23288#issuecomment-1380811808

### Would you like to implement a fix?

None

**Comment by YakDriver** [comment]
I agree that this is a great idea but note that the testing framework does run a refresh after the apply. So, for a 1 step test, it plans, applies, and then refreshes. This catches many, many drift situations. While drift is annoying and a problem, the incidence of drift are actually very low given the amount applies and refreshes happening, thanks in large part to the testing framework having that final refresh.

There are many places in the tests where we apply over and over to check for this exact problem. Unfortunately, even where we apply many times in a test, it often doesn't reproduce what people are seeing in their environments. I'm not sure why this would be. In cases, I've taken great lengths to exactly reproduce the configuration, and then apply over and over in the test, and the test always comes back clean. See this test, for example:

https://github.com/hashicorp/terraform-provider-aws/blob/d7c92b46531e0c75053ff9511a8fe3f60555103b/internal/service/iam/role_test.go#L178-L313

**Comment by nitrocode** [comment]
Thank you for the in-depth response @YakDriver.

Feel free to close this if this no longer has value. 

I know it's been some time now since our last discussion. Has anything changed since? Were you able to track down any other differences between the test environment and real world to improve the test suite or is it more or less the same? 

Thanks
